### PR TITLE
HOMS-108 Add mass printing

### PR DESCRIPTION
**Pull request by sugarfree1**

### CODE REVIEW

**Comment by sugarfree1** [comment]
Updated

File: app/controllers/imprint/prints_controller.rb [hunk 1]
```diff
@@ -1,15 +1,37 @@
 module Imprint
   class PrintsController < ApplicationController
+    include ::Validations
+    extend Imprint::Mixin
+
+    inject['imprint_adapter']
```
Comment (flash-gordon) [inline app/controllers/imprint/prints_controller.rb]: this actually can be simplified as
```ruby
inject['imprint_adapter', 'services.order_printing_service']
```

File: config/application.rb [hunk 2]
```diff
@@ -47,6 +51,12 @@ class Application < Rails::Application
 
     config.after_initialize do
       I18n.locale = I18n.default_locale = (config.app.locale.fetch(:code) || :en).to_sym
+
+      Dir[Rails.root.join('lib/imprint/*.rb')].each { |f| require f }
```
Comment (flash-gordon) [inline config/application.rb]: these manual require calls are rather nasty

File: lib/imprint/services/order_printing_service.rb [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+          raise OrderMultipleTypesException
+        end
+
+        orders
+      end
+
+      def prepare_filter_params(current_user, parameters)
+        user_ids = parameters[:filter] ? parameters[:user_id] : [current_user.id, 'empty']
+
+        parameters.permit(:state, :order_type_id, :archived,
```
Comment (flash-gordon) [inline lib/imprint/services/order_printing_service.rb]: I believe we should use dry-v for that

**Comment by sugarfree1** [comment]
Updated

**Comment by sugarfree1** [comment]
Updated

**Comment by flash-gordon** [comment]
Bite my shiny metal ass!

**Comment by sugarfree1** [comment]
It's passing now!

**Pull request merged**